### ISSUE TITLE
Bug: `sam deploy --guided` makes public deployment bucket

### ISSUE THREAD

**Issue by nickls** [aws/aws-sam-cli#4704]
When running `sam deploy --guided` it makes an S3 deployment bucket that has public access turned on. This is unnecessary and a (minor) security risk.

### Description:
![image](https://user-images.githubusercontent.com/363928/218536430-8a171748-31ce-4186-9151-5dffc21abe6f.png)

### Steps to reproduce:
Run `sam deploy --guided` check S3 console.

### Observed result:
S3 bucket is publicly accessible:
![image](https://user-images.githubusercontent.com/363928/218536430-8a171748-31ce-4186-9151-5dffc21abe6f.png)

### Expected result:
S3 bucket is NOT publicly accessible:
![image](https://user-images.githubusercontent.com/363928/218537280-c8464bd9-bf60-48c3-85d0-41a423cf6e06.png)

**Comment by jfuss** [comment]
@nickls What version of SAM CLI are you running?

Looking, at the template we deploy, Public should be blocked on the bucket: https://github.com/aws/aws-sam-cli/blob/f8e3368ebcc541f5c813a82cc0b0db43a35db391/samcli/lib/bootstrap/bootstrap.py#L65

**Comment by mattgkmart** [comment]
Can confirm - NOT public with SAM cli 1.72. I don't think this is an issue.

![image](https://user-images.githubusercontent.com/111095066/218637405-c42f089e-6da4-4a4d-b684-40818ccc9a73.png)

**Comment by mattgillard** [comment]
Can confirm NOT public with sam cli 1.72. I don't think this is an issue.

![image](https://user-images.githubusercontent.com/6270663/218637924-83fe7238-6fba-4114-ac12-75233e52c8e8.png)

**Comment by nickls** [comment]
@jfuss & @mattgillard  -- I'm on `SAM CLI, version 1.36.0`

**Comment by jfuss** [comment]
@nickls You will need to delete the current "managed stack" and s3 bucket. We don't currently update the managed stack after deployment (which is a separate issue we are aware of).

**Comment by nickls** [comment]
@jfuss  -- I deleted the stack and recreated it. Bucket is now not public. 🚀

![image](https://user-images.githubusercontent.com/363928/218837443-9940c554-e60e-4dcb-b541-1eda7e6252c6.png)

**Comment by jfuss** [comment]
@nickls Thanks for confirming and reaching out!
